File: tiq-e/trees/4/main.py
from collections import deque
from time import perf_counter
from typing import List


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        res = []
        queue = deque([root])
        while queue:
            row = []
            for _ in range(len(queue)):
                root = queue.popleft()
                row.append(root.val)
                if root.left:
                    queue.append(root.left)
                if root.right:
                    queue.append(root.right)
            res.append(row)
        return res


root = TreeNode(5, None, None)

start = perf_counter()
result = Solution().levelOrder(root)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/trees/4/main.go
package main

import (
	"fmt"
	"time"
)

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	var result [][]int
	if root == nil {
		return result
	}
	level := []*TreeNode{root}
	for len(level) > 0 {
		var tmp1 []*TreeNode
		var tmp2 []int
		for _, c := range level {
			if c != nil {
				tmp2 = append(tmp2, c.Val)
				tmp1 = append(tmp1, c.Left)
				tmp1 = append(tmp1, c.Right)
			}
		}
		if len(tmp2) != 0 {
			result = append(result, tmp2)
		}
		level = tmp1

	}
	return result
}

func main() {
	var root = TreeNode{
		Val:   5,
		Left:  nil,
		Right: nil,
	}

	var start = time.Now()
	var result = levelOrder(&root)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
